sqlglot/parser.py

```diff
@@ -558,12 +546,12 @@ class Parser(metaclass=_Parser):
         TokenType.ARROW: lambda self, this, path: self.expression(
             exp.JSONExtract,
             this=this,
-            expression=parse_json_path(path),
+            expression=self.dialect.parse_json_path(path),
```

Comment:
why don't we just call self.parse_json_path?

Reply:
Let's discuss this in https://github.com/tobymao/sqlglot/pull/2911#discussion_r1476873600 since they're related